src/Visualization.jl

```diff
@@ -28,27 +28,52 @@ function Makie.plot!(f::Makie.GridPosition, tn::TensorNetwork{A}; labels = false
     # TODO recognise them by using `DeltaArray` or `Diagonal` representations
     copytensors = findall(t -> haskey(t.meta, :dual), tensors(tn))
 
-    kwargs[:node_size] = [max(15, log2(size(tensors(tn, i)) |> prod)) for i in 1:nv(graph)]
-    kwargs[:node_marker] = [i ∈ copytensors ? :diamond : :circle for i in 1:length(tensors(tn))]
-    kwargs[:node_color] = [i ∈ copytensors ? :black : :white for i in 1:length(tensors(tn))]
+    oinds = openinds(tn)
+    ghostnodes = []
+    while !isempty(oinds)
+        for ind in oinds
+            tensor = links(ind)[1]
+            tensor_oinds = filter(id -> id.name ∈ tensor.labels, oinds)
+
+            add_vertex!(graph)
+            add_edge!(graph, pos[tensor], nv(graph))
+
+            push!(ghostnodes, nv(graph))
+            filter!(id -> id ∉ tensor_oinds, oinds)
+        end
+    end
```

Comment:
In general, I don't see why the `while` loop is needed, or why the need for filtering. I think the following code is cleaner and much more straightforward.
```suggestion
    ghostnodes = map(openinds(tn)) do ind
        add_vertex!(graph)

        node = nv(graph) # TODO is this the best way to get the id of the newly created node?
        tensor = only(links(ind))
        add_edge!(graph, node, pos[tensor])

        return node
    end
```

Reply:
Okey, this can also work. The `while` and `for` was there since that worked better when I created random tensors. I agree that this is better now.

Reply:
I don't think this is a good idea. This index union assumes both indices will be connected to the same tensor, but its something that we don't know yet.

I think leaving them separate is much better.

Reply:
Okay, I joined the labels because we do this when we have tensors that share indices. I will change the code to work as you say.